pkg/pillar/cmd/zedagent/handlenetworkinstance.go

```diff
@@ -761,6 +761,56 @@ func handleAppFlowMonitorDelete(ctxArg interface{}, key string,
 	log.Infof("handleAppFlowMonitorDelete(%s) done\n", key)
 }
 
+func handleAppVifIPTrigModify(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleAppVifIPTrigModify(%s)\n", key)
+	ctx := ctxArg.(*zedagentContext)
+	trig := cast.CastVifIPTrig(statusArg)
+	if trig.Key() != key {
+		log.Errorf("handleAppVifIPTrigModify key mismatch %s vs %s; ignored\n",
+			key, trig.Key())
+		return
+	}
+	findVifAndTrigAppInfoUpload(ctx, trig.Key())
+}
+
+func handleAppVifIPTrigDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleAppVifIPTrigDelete(%s)\n", key)
+	trig := cast.CastVifIPTrig(statusArg)
+	if trig.Key() != key {
+		log.Errorf("handleAppVifIPTrigModify key mismatch %s vs %s; ignored\n",
+			key, trig.Key())
+		return
+	}
+	log.Infof("handleAppVifIPTrigDelete(%s) done\n", key)
+}
+
+func findVifAndTrigAppInfoUpload(ctx *zedagentContext, macAddr string) {
+	sub := ctx.getconfigCtx.subAppInstanceStatus
+	items := sub.GetAll()
+
+	for _, st := range items {
+		var aiFound bool
+		aiStatus := cast.CastAppInstanceStatus(st)
+		for _, ulStatus := range aiStatus.UnderlayNetworks {
```

Comment:
Is this If block better be a member function of aiStatus??
 
aiUsesMac := aiStatus.HasUnderlayNetworkWithMacAddr(macAddr)
 ...

Reply:
make sense. will do.